editor/src/clj/editor/gui.clj

```diff
@@ -501,9 +501,13 @@
       (validation/prop-error :fatal node-id prop-kw (partial validation/prop-id-duplicate? id-counts) prop-value)))
 
 ;; SDK api
-(defn prop-resource-error [node-id prop-kw prop-value prop-name]
-  (or (validation/prop-error :fatal node-id prop-kw validation/prop-nil? prop-value prop-name)
+(defn prop-resource-error
+  ([node-id prop-kw prop-value prop-name]
+    (or (validation/prop-error :fatal node-id prop-kw validation/prop-nil? prop-value prop-name)
       (validation/prop-error :fatal node-id prop-kw validation/prop-resource-not-exists? prop-value prop-name)))
+  ([node-id prop-kw prop-value prop-name resource-ext]
+    (or (prop-resource-error node-id prop-kw prop-value prop-name)
+      (validation/prop-error :fatal node-id prop-kw validation/prop-resource-ext? prop-value resource-ext prop-name))))
```

Comment:
It's public API + we don't need it in all the cases, so I do this new checking optional.